src/hotspot/share/gc/shared/referenceProcessor.hpp

```diff
@@ -253,6 +253,9 @@ class ReferenceProcessor : public ReferenceDiscoverer {
   DiscoveredList* _discoveredFinalRefs;
   DiscoveredList* _discoveredPhantomRefs;
 
+  // Workers to process the discovered non-strong references.
+  WorkerThreads* _workers;
```

Comment:
Rather than adding a WorkerThreads member at construction time, it seems better to me to add an argument to ReferenceProcessor::run_task.

Reply:
When a ref-processor is instantiated, the caller specifies #worker to use for ref-processing. Since #workers and the worker-pool are related, I think it's conceptually cleaner that they are set in the same place.

Using the arg approach would mean that an extra arg has to be passed through `run_task` <- (`process_soft_weak_final_refs` | `process_final_keep_alive` | `process_phantom_refs`) <- `process_discovered_references`; this seems harder to follow where `workers` comes from.